article/app/model/AutomaticFilters.scala

```diff
@@ -0,0 +1,26 @@
+package model
+
+import common.GuLogging
+import model.TopMentionEntity.TopMentionEntity
+
+object AutomaticFilters extends GuLogging {
+
+  def getAutomaticFilter(filter: Option[String]): Option[(TopMentionEntity, String)] = {
+    filter.flatMap { f =>
+      val filterEntity = f.split(":")
+      if (filterEntity.length == 2) {
+        val entityType = TopMentionEntity.withNameOpt(filterEntity(0).toUpperCase)
+        if (entityType.isEmpty) {
+          log.error(s"automaticFilter query parameter entity ${filterEntity(0)} is invalid")
```

Comment:
this could also happen when that particular filter doesn't exist in memory for any reason, maybe the error could be a warning and say something like "Could not find top mentions data for provided entity [${filterEntity(0)}]"